### PR TITLE
Merge CCCL service builder and deployer

### PR DESCRIPTION
**Pull request by richbrowne**
Changes necessary to deploy a CCCL configuration.  This implements the apply_config() functionality.

### CODE REVIEW

File: f5_cccl/api.py [hunk 1]
```diff
@@ -30,27 +36,35 @@ class F5CloudServiceManager(object):
     under its control.
     """
 
-    def __init__(self, bigip, partition, prefix=None):
+    def __init__(self, hostname, username, password, partition, prefix=None,
+                 port=443, token=None):
```
Comment (edarzins) [inline f5_cccl/api.py]: arg list doesn't match docstring that follows.

File: f5_cccl/resource/ltm/monitor/http_monitor.py [hunk 2]
```diff
@@ -38,7 +30,15 @@ class HTTPMonitor(Monitor):
...
     """
-    monitor_schema_kvps = None
+    http_properties = dict(interval=5,
+                           timeout=16,
+                           send="GET /\\r\\n",
+                           recv="")
+
+    def __init__(self, name, partition, **kwargs):
+        super(HTTPMonitor, self).__init__(name, partition, **kwargs)
+        for key in ['send', 'recv']:
```
Comment (edarzins) [inline f5_cccl/resource/ltm/monitor/http_monitor.py]: Shouldn't you iterate through http_properties?

Comment (edarzins) [inline f5_cccl/resource/ltm/monitor/http_monitor.py]: Never mind. interval and timeout are properties of the base class Monitor.

File: f5_cccl/service/manager.py [hunk 3]
```diff
@@ -14,3 +15,187 @@
...
+            create_tasks = self._create_resources(create_tasks)
+
+            update_tasks = self._update_resources(update_tasks)
+
+            delete_tasks = self._delete_resources(delete_tasks)
+
+            tasks_remaining = (
+                len(create_tasks) + len(update_tasks) + len(delete_tasks))
+
+            # Did the tqsk queue shrink?
```
Comment (edarzins) [inline f5_cccl/service/manager.py]: typo: task

File: f5_cccl/service/manager.py [hunk 4]
```diff
@@ -14,3 +15,187 @@
...
+            self._get_monitor_tasks(desired_config))
+
+        create_tasks = create_monitors + create_pools + create_virtuals
+        update_tasks = update_monitors + update_pools + update_virtuals
+        delete_tasks = delete_virtuals + delete_pools + delete_monitors
+
+        taskq_len = len(create_tasks) + len(update_tasks) + len(delete_tasks)
+
+        finished = False
+        while not finished:
```
Comment (edarzins) [inline f5_cccl/service/manager.py]: Please add a comment before this loop for how this functions -- this is important!

File: f5_cccl/service/manager.py [hunk 5]
```diff
@@ -14,3 +15,187 @@
...
+        ]
+
+        return (create_list, update_list, delete_list)
+
+    def _create_resources(self, create_list):
+        """Iterate over the resources and call create method."""
+        retry_list = list()
+        for resource in create_list:
+            try:
+                print("Creating %s..." % resource.name)
```
Comment (edarzins) [inline f5_cccl/service/manager.py]: "print" is okay for now, but velcro will need these to be logged (There's already a story for logging.)

File: test/f5_cccl/resource/ltm/monitor/test_fn_monitor.py [hunk 6]
```diff
@@ -35,7 +35,7 @@
 
 from icontrol.exceptions import iControlUnexpectedHTTPError
 
-import monitor_schemas
+# import monitor_schemas
```
Comment (edarzins) [inline test/f5_cccl/resource/ltm/monitor/test_fn_monitor.py]: remove this line

**Comment by richbrowne** [comment]
Thanks Erik for looking at this -- I know it was a big commit.  I have made the changes you suggested.  Appreciate it.

File: f5_cccl/api.py [hunk 7]
```diff
@@ -40,10 +40,15 @@ def __init__(self, hostname, username, password, partition, prefix=None,
                  port=443, token=None):
         """Initialize an instance of the F5 CCCL service manager.
 
-        :param bigip:  Interface to BIG-IP device, f5-common-python
-        ManagementRoot (f5.bigip.ManagementRoot)
+        :param hostname: BIG-IP hostname or ip address.
+        :param username: Access BIG-IP as user
+        :param password: User password
+        :param por
```
Comment (edarzins) [inline f5_cccl/api.py]: Delete this line

File: f5_cccl/api.py [hunk 8]
```diff
@@ -40,10 +40,15 @@ def __init__(self, hostname, username, password, partition, prefix=None,
...
         :param partition: Name of BIG-IP partition to manage.
-        :param prefix: Optional string to prepend to resource names.
+        :param prefix: Optional string to prepend to resource names
+        (default: None).
+        :param part: Port to use for connection (default: 443)
```
Comment (edarzins) [inline f5_cccl/api.py]: Change 'part' to 'port'.

Comment (richbrowne) [inline f5_cccl/api.py]: Ok, I fixed these.  I was being distracted by my daughter.

**Pull request merged**